### ISSUE TITLE
Creating a Request refrencing a Person

### ISSUE THREAD

**Issue by RobertOsterberg** [code4me/4me-sdk-csharp#56]
Hi again, i am making some progress using the SDK and was wondering if you could show an example of Creating a Request refrencing a Person.
RequestedBy and RequestedFor properties on the request. 
I have all the data as primaryEmail and want to find/refrence those people and create the request.

How do i filter for one person
code so far

List<Person> people = client.People.Get(
                                    new FilterCollection()
                                    {
                                         new Filter("Name", FilterCondition.Equality, "Robert Österberg")
                                    }
                                    );

**Comment by klaasvandeweerdt** [comment]
Hello,

You will need a few references to create a request. As you mentioned the Requested For and Requested By, but also the ServiceInstance. You may need the Team and Request Template references as well.

The API endpoint needs the identifiers for all those reference fields, nothing more.
If you know those identifiers then you can reference them without searching for them, which reduces the number of API calls.

Have a look at the [documentation](https://developer.4me.com/v1/requests/) for the required fields.

Your code snippet is the correct way to search for people, although I would try to use the ```PrimaryEmail``` or ```SourceID``` instead.

The first step is to get those reference values. Using ```System.Linq.First()``` allows you to grab the first item from the result collection and it will throw an exception when no item was returned.
```csharp
var requestedBy = client.People.Get(new Filter("Name", FilterCondition.Equality, "Klaas")).First();
var requestedFor = client.People.Get(new Filter("PrimaryEmail", FilterCondition.Equality, "user@email.com")).First();
var serviceInstance = new ServiceInstance() { ID = 1234 };
```
Optional
```csharp
var team = new Team() { ID = 1234};
var template = new RequestTemplate() { ID = 1234 };
```

Once you have those you can create the request.
```csharp
var request = new Request()
{
    RequestedBy = requestedBy,
    RequestedFor = requestedFor,
    ServiceInstance = serviceInstance,
    Status = RequestStatus.Assigned,
    //Team = team,
    //Template = template,
    Subject = "An incident",
    Category = RequestCategory.Incident,
    Impact = RequestImpact.Low,
    //Note = "A note",
    //InternalNote = "An internal note"
};
request = client.Requests.Insert(request);
```

**Comment by RobertOsterberg** [comment]
thanks so much i will try it out

**Comment by RobertOsterberg** [comment]
The strange thing is that the applications just hangs and nothing is happening, not even an exception. I tried to look in the System in logs but can`t find anything.  No problems setting the client.
Any idea
line 130
![image](https://user-images.githubusercontent.com/18688250/190858880-23eef480-fcb4-4fd6-8894-7f307713f100.png)

**Comment by klaasvandeweerdt** [comment]
The API endpoints only allow 20 requests per 2 seconds, so after every call, the SDK calculates a sleep period. There was a bug in the sleep period calculation, which was resolved in version 2.3.0.2. Make sure to use the latest version.

**Comment by RobertOsterberg** [comment]
Great thanks again Klaas
